src/idle_inhibitor/wayland/mod.rs

```diff
@@ -74,10 +86,63 @@ impl WaylandIdleInhibitor {
             _registry: registry,
             data: AppData::default(),
         };
-        obj.roundtrip()?;
+        obj.initialize()?;
         Ok(obj)
     }
 
+    fn init_buffer(&mut self) {
+        let mut file = tempfile::tempfile().unwrap();
+        let shm = self.data.shm.as_ref().expect("WlShm global not initialized");
+        let width = 1;
+        let height = 1;
+        let stride = width * 4;
+        let pool_size = height * stride * 2;
+        let pool = shm.create_pool(file.as_fd(), pool_size, &self.qhandle, ());
+        let buffer = pool.create_buffer(
+            0,
+            width,
+            height,
+            stride,
+            Format::Argb8888,
+            &self.qhandle,
+            (),
+        );
+        let _ = file.write(b"\xff\xff\xff\xff");
```

Comment:
Causes an error because the buffer is not big enough. Also, it create a white coloured background.

```suggestion
        let _ = file.write(b"\x00\x00\x00\x00\x00\x00\x00\x00");
```